🐛 Improve `env.rc` and `create_cloud_conf.sh` templates

skatsaounis opened the pull request:
**What this PR does / why we need it**:

It fixes the following erroneous situations:
1. Scripts are run from macOS where base64 supports `--break` instead of `--wrap`
2. For security reasons OpenStack password is present in `OS_PASSWORD` environment variable rather than clouds.yaml

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:
<!-- Put an "X" character inside the brackets of each completed task. Some may be optional depending on the PR. -->

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

mdbooth commented on templates/env.rc:
```diff
@@ -63,17 +63,30 @@ CAPO_OPENSTACK_CLOUD_YAML_CONTENT=$(cat "${CAPO_CLOUDS_PATH}")
-        else
-                yq e .$2 $1
-        fi
+  if [[ ${CAPO_YQ_VERSION} == *"version 1."* || ${CAPO_YQ_VERSION} == *"version 2."* || ${CAPO_YQ_VERSION} == *"version 3."* ]]; then
```
Is there a reason for adding the dots here? If so, is there any reason not to do it elsewhere?

skatsaounis replied:
No, not really. I just reverted the change

mdbooth reviewed (changes requested):
This looks ok to me, and presumably very useful if you're running it on a Mac!

Please could you remove all the whitespace and incidental changes? They make the change itself much harder to see and will also make future git spelunking harder by adding non-functional noise. If the whitespace changes are important we could consider them in a separate PR, though.

Thanks!

seanschneeweiss commented:
/ok-to-test

seanschneeweiss commented:
@skatsaounis Would be great if you could address the changes proposed by @mdbooth.
Let us know if something is unclear of if you need assistance for completing this PR.

skatsaounis commented:
Hi both @seanschneeweiss and @mdbooth! Sorry for such a huge delay in addressing the PR review comments.

Basically, I just removed what is not related to the changes that are fixing the issue with macOS and fetching the OS_PASSWORD from an environment variable.

I think now the PR is much more clear and ready to be merged if you agree to do so.

Once again, sorry for the delay and thank you for taking time to consider it and review it 🍻

tobiasgiese commented on templates/env.rc:
```diff
@@ -70,10 +70,23 @@ yqNavigating(){
 
+b64encode(){
+  # Check if wrap is supported. Otherwise, break is supported.
+  echo | base64 --wrap=0 > /dev/null 2>&1
+  if [ $? -eq 0 ]; then
```
nit: we can simplify this :)

```suggestion
  if echo | base64 --wrap=0 &> /dev/null; then
```

tobiasgiese reviewed (approved):
One nit, but all in all LGTM

skatsaounis commented:
@tobiasgiese thanks for the suggestion, which it makes sense. I just applied with a separate commit through the UI.

tobiasgiese commented:
> I just applied with a separate commit through the UI.

Can you please squash the commits? Prow will rebase this branch into master during merge, thus commits must be squashed :)

mdbooth reviewed (commented):
/lgtm

jichenjc commented:
/approve

mdbooth commented:
/lgtm

skatsaounis commented:
/remove-lifecycle rotten

skatsaounis commented:
/remove-hold
